### PR TITLE
Cypress test cases

### PR DESCRIPTION
**Pull request by OSKovalchuk**

### CODE REVIEW

File: cypress/e2e/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+
+1. Clone this repository to your local machine.
+
+   ```bash
+   git clone https://github.com/OSKovalchuk/Final-test.-Cypress.git
+
+
+2. Install the necessary dependencies.
+   npm run
```
Comment (Vitaliitymashkov) [inline cypress/e2e/README.md]: Run doesn't install dependencies.

File: cypress/e2e/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+3. Running Cypress Tests
+    To run the Cypress tests, follow these steps:
+
+    Open a terminal in the project directory.
+
+    Run Cypress.
```
Comment (Vitaliitymashkov) [inline cypress/e2e/README.md]: It is GUI mode.

Also non-GUI mode can be created.
npx cypress run

File: cypress/e2e/Test Cases/Test Case-1-2.cy.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  
+      // Verify the URL is changed to the inventory page
+      cy.url().should('include', '/inventory.html');
+  
+      // Open the sidebar to check for the logout element
+      cy.get('.bm-burger-button').click(); // Assuming you are using BurgerMenu for sidebar
+      cy.get('#logout_sidebar_link').should('be.visible');
+      
+      //Logout from the application
+      cy.get('#logout_sidebar_link').click();
```
Comment (Vitaliitymashkov) [inline cypress/e2e/Test Cases/Test Case-1-2.cy.js]: TC1 doesn't require logout :)

File: cypress/e2e/Test Cases/Test Case-3.cy.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  
+      // Enter invalid username and password
+      cy.get('input[data-test="username"]').type('invalid_user');
+      cy.get('input[data-test="password"]').type('invalid_password');
+  
+      // Click the login button
+      cy.get('input[type="submit"]').click();
+  
+      // Verify the error label is visible with the correct text
+      //cy.get('h3[data-test="error"]').should('be.visible').and('have.text', 'Username and password do not match any user in this service.');
```
Comment (Vitaliitymashkov) [inline cypress/e2e/Test Cases/Test Case-3.cy.js]: Why an assertion is disabled? Test is not checking for the label.

File: cypress/e2e/Test Cases/Test Case-7.cy.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      cy.get('input[data-test="firstName"]').type('John');
+      cy.get('input[data-test="lastName"]').type('Doe');
+      cy.get('input[data-test="postalCode"]').type('12345');
+      cy.get('input[data-test="continue"]').click();
+  
+      // Finish the checkout
+      cy.get('button[data-test="finish"]').click();
+  
+      // Verify the URL is changed to the thank you page
+      cy.url().should('include', '/checkout-complete.html');
```
Comment (Vitaliitymashkov) [inline cypress/e2e/Test Cases/Test Case-7.cy.js]: Test could be more advanced - checking the selected item, price, tax, total.

**Review by Vitaliitymashkov — APPROVED** [review]
Approved. No CI pipelines = no extra points.

File: cypress/e2e/Test Cases/Test Case-1-2.cy.js (hunk 3 above)
Comment (OSKovalchuk) [reply cypress/e2e/Test Cases/Test Case-1-2.cy.js]: There test case 1 and test case 2 together

Comment (Vitaliitymashkov) [reply cypress/e2e/Test Cases/Test Case-1-2.cy.js]: Decided to split 2 test cases.

File: cypress/e2e/Test Cases/Test Case-3.cy.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+// cypress invalidLogin.spec.js
+
+describe('Invalid Login Test', () => {
+    it('Valid (standard_user) user cannot log in with invalid credentials', () => {
+      // Visit the login page
+      cy.visit('https://www.saucedemo.com/');
```
Comment (Vitaliitymashkov) [inline cypress/e2e/Test Cases/Test Case-3.cy.js]: Move baseUrl into config file
https://docs.cypress.io/guides/references/configuration
https://filiphric.com/cypress-basics-using-baseurl

File: cypress/e2e/Test Cases/Test Case-4.cy.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+// cypress/integration/lockedOutUser.spec.js
+
+describe('Locked-Out User Test', () => {
+    it('Locked-out user cannot log in with valid credentials', () => {
+      // Visit the login page
```
Comment (Vitaliitymashkov) [inline cypress/e2e/Test Cases/Test Case-4.cy.js]: All comments which are obvious are unnecessary. 

example
checkoutPage.sortItemsListByOrder();
Below there an obvious comment which never should get into the code.
//Checkout Page - sort Items by Order

File: cypress/e2e/Test Cases/Test Case-7.cy.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+      // Open the cart
+      cy.get('.shopping_cart_link').click();
+  
+      // Click on the checkout button
+      cy.get('button[data-test="checkout"]').click();
+  
+      // Enter checkout information and continue
+      cy.get('input[data-test="firstName"]').type('John');
+      cy.get('input[data-test="lastName"]').type('Doe');
+      cy.get('input[data-test="postalCode"]').type('12345');
```
Comment (Vitaliitymashkov) [inline cypress/e2e/Test Cases/Test Case-7.cy.js]: Need to learn "clean code".
https://github.com/ryanmcdermott/clean-code-javascript

const POSTAL_CODE = '12345';
cy.get('input[data-test="postalCode"]').type(POSTAL_CODE);

File: cypress/e2e/Test Cases/Test Case-8.cy.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+describe('Responsive Login Test', () => {
+    it('Valid user can login when screen width is less than 1060px', () => {
+      // Set the viewport size to less than 1060px width
+      cy.viewport(1000, 800);
+  
+      // Visit the login page
+      cy.visit('https://www.saucedemo.com/');
+  
+      // Log in with a valid username and password
```
Comment (Vitaliitymashkov) [inline cypress/e2e/Test Cases/Test Case-8.cy.js]: Should be in the before hook
https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests
https://filiphric.com/cypress-basics-before-beforeeach-after-aftereach

**Pull request merged**